src/Types/ClickToPayHelpers.res

```diff
@@ -0,0 +1,961 @@
+
+let srcUiKitScriptSrc = "https://src.mastercard.com/srci/integration/components/src-ui-kit/src-ui-kit.esm.js"
+let srcUiKitCssHref = "https://src.mastercard.com/srci/integration/components/src-ui-kit/src-ui-kit.css"
+
+let recognitionTokenCookieName = "__mastercard_click_to_pay"
+let manualCardId = "click_to_pay_manual_card"
+let savedCardId = "click_to_pay_saved_card_"
+
+let orderIdRef = ref("")
+
+type element = {
+  mutable innerHTML: string,
+  appendChild: CommonHooks.element => unit,
+  removeChild: CommonHooks.element => unit,
+  replaceChildren: unit => unit,
+  children: array<CommonHooks.element>,
+}
+type elementDocument
+@val external myDocument: elementDocument = "document"
+@send external getElementById: (elementDocument, string) => Nullable.t<element> = "getElementById"
+
+open Window
+let clickToPayWindowRef: ref<Nullable.t<window>> = ref(Nullable.null)
+
+@send external closeWindow: window => unit = "close"
+@val external windowOpen: (string, string, string) => Nullable.t<window> = "open"
+
+let handleCloseClickToPayWindow = () => {
+  switch clickToPayWindowRef.contents->Nullable.toOption {
+  | Some(window) => {
+      window->closeWindow
+      clickToPayWindowRef.contents = Nullable.null
+    }
+  | None => ()
+  }
+}
+
+let handleOpenClickToPayWindow = () => {
+  clickToPayWindowRef.contents = windowOpen("", "ClickToPayWindow", "width=480,height=600")
+}
+
+// Global window extensions
+type mastercardCheckoutServices
+
+@val
+external mastercardCheckoutServices: mastercardCheckoutServices =
+  "window.MastercardCheckoutServices"
+
+// Response types
+type responsePayloadStatus = COMPLETE | CANCEL | PAY_V3_CARD | ERROR
+
+type responsePayload = {
+  status: responsePayloadStatus,
+  payload: JSON.t,
+}
+
+type cardBrand = [
+  | #visa
+  | #mastercard
+  | #discover
+  | #amex
+]
+
+type authenticationPreferences = {payloadRequested: [#AUTHENTICATED | #NON_AUTHENTICATED]}
+
+type paymentOption = {
+  dpaDynamicDataTtlMinutes: int,
+  dynamicDataType: [#CARD_APPLICATION_CRYPTOGRAM_SHORT_FORM | #NONE],
+}
+
+type transactionAmount = {
+  transactionAmount: float,
+  transactionCurrencyCode: string,
+}
+
+type dpaTransactionOptions = {
+  dpaLocale: string,
+  authenticationPreferences: authenticationPreferences,
+  paymentOptions: array<paymentOption>,
+  transactionAmount: transactionAmount,
+  acquirerBIN: string,
+  acquirerMerchantId: string,
+  merchantCategoryCode: string,
+  merchantCountryCode: string,
+}
+
+type dpaData = {dpaName: string}
+
+type params = {
+  srcDpaId: string,
+  dpaData: dpaData,
+  dpaTransactionOptions: dpaTransactionOptions,
+  cardBrands: array<string>,
+  recognitionToken?: string,
+  checkoutExperience: [#WITHIN_CHECKOUT | #PAYMENT_SETTINGS],
+}
+
+type country = {
+  code: string,
+  countryISO: string,
+}
+
+let defaultCountry: country = {code: "", countryISO: ""}
+
+// Cookie helpers
+@val external document: {..} = "document"
+
+let setLocalStorage = (~key: string, ~value: string) => {
```

Comment:
Hey @seekshiva, for ClickToPay should we store the recognitionToken in the cookies or in localStorage? Both scenarios will work in Chrome browser however cause issue in Safari

Reply:
what works in safari?